evil-search.el

```diff
@@ -191,10 +191,10 @@ of the buffer."
            result)
        (setq result (funcall search-fun string bound
                              ,(if wrap t 'noerror) count))
-       (when (and ,wrap (null result))
+       (when (and ,wrap (null result) (> start 1))
```

Comment:
This should actually be `(> start (point-min))` since the buffer may be narrowed.

Also this only considers searching in the forward direction, we should also consider searching in the backward direction, i.e. when `forward` is `nil`.